File: src/frontend/src/app/core/models/syllabus/point-limit.ts
import { KindOfSubject } from "../../enums/subject/kind-of-subject.enum";
import { ModuleType } from "../../enums/subject/module-type.enum";
import { TypeOfSubject } from "../../enums/subject/type-of-subject.enum";
import { ModelBase } from "../model-base";

export class PointLimit extends ModelBase {
    moduleType: ModuleType = ModuleType.FieldOfStudy;
    kindOfSubject?: KindOfSubject;
    typeOfSubject?: TypeOfSubject;
    points: number = 0;
}

File: src/backend/SyllabusManager.Data/Models/Syllabuses/PointLimit.cs
using SyllabusManager.Data.Enums.Subjects;

namespace SyllabusManager.Data.Models.Syllabuses
{
    public class PointLimit : ModelBase
    {
        public int Points { get; set; } = 0;
        public ModuleType ModuleType { get; set; }
        public KindOfSubject KindOfSubject { get; set; }
    }
}
